File: python/problem1047.py
entrada = list(map(int, input().split()));

a = entrada[0];
b = entrada[1];
c = entrada[2];
d = entrada[3];

minInicio = (a * 60) + b;
minFim = (c * 60) + d;

dia = 24 * 60;

if minInicio > minFim:
	tempo = dia - (minInicio - minFim);
	horas =  tempo // 60;
	minutos = tempo % 60;
elif minInicio < minFim:
	tempo = minFim - minInicio;
	horas = tempo // 60;
	minutos = tempo % 60;
else:
	horas = 24;
	minutos = 0;
print("O JOGO DUROU %d HORA(S) E %d MINUTO(S)" % (horas, minutos))

File: java/Problem1047.java
import java.util.Scanner;

public class Problem1047 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int inicio = sc.nextInt() * 60;
        inicio += sc.nextInt();

        int fim = sc.nextInt() * 60;
        fim += sc.nextInt();

        int total;

        if(inicio < fim){
            total = fim - inicio;
        } else if(inicio > fim){
            total = fim + (24 * 60) - inicio;
        } else {
            total = (24 * 60);
        }

        System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", total / 60, total % 60 );

        sc.close();
    }
}